spotinst/commons/administration_org_programmatic_user/fields_spotinst_administration_org_programmatic_user.go

```diff
@@ -0,0 +1,312 @@
+		},
+		func(resourceObject interface{}, resourceData *schema.ResourceData, meta interface{}) error {
+			orgProgUserWrapper := resourceObject.(*commons.OrgProgUserWrapper)
+			orgProgUser := orgProgUserWrapper.GetOrgProgUser()
+			if v, ok := resourceData.GetOk(string(Description)); ok && v != "" {
+				orgProgUser.SetDescription(spotinst.String(resourceData.Get(string(Description)).(string)))
+			}
+			return nil
+		},
+		nil,
+	)
+
+	fieldsMap[Policies] = commons.NewGenericField(
+		commons.AdministrationOrgProgrammaticUser,
+		Policies,
+		&schema.Schema{
+			Type:     schema.TypeSet,
+			Optional: true,
+			ForceNew: true,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					string(PolicyAccountIds): {
+						Type:     schema.TypeList,
+						Required: true,
+						Elem:     &schema.Schema{Type: schema.TypeString},
+					},
+					string(PolicyId): {
+						Type:     schema.TypeString,
+						Required: true,
+					},
+				},
+			},
+		},
+		func(resourceObject interface{}, resourceData *schema.ResourceData, meta interface{}) error {
+			orgProgUserWrapper := resourceObject.(*commons.OrgProgUserWrapper)
+			orgProgUser := orgProgUserWrapper.GetOrgProgUser()
+			var result []interface{} = nil
+			if orgProgUser.Policies != nil {
+				policies := orgProgUser.Policies
+				result = flattenPolicies(policies)
+			}
+			if result != nil {
+				if err := resourceData.Set(string(Policies), result); err != nil {
+					return fmt.Errorf(string(commons.FailureFieldReadPattern), string(Policies), err)
+				}
+			}
+			return nil
+		},
+		func(resourceObject interface{}, resourceData *schema.ResourceData, meta interface{}) error {
+			orgProgUserWrapper := resourceObject.(*commons.OrgProgUserWrapper)
+			orgProgUser := orgProgUserWrapper.GetOrgProgUser()
+			if v, ok := resourceData.GetOk(string(Policies)); ok {
+				if policies, err := expandPolicies(v); err != nil {
+					return err
+				} else {
+					orgProgUser.SetPolicies(policies)
+				}
+			}
+			return nil
+		},
+		func(resourceObject interface{}, resourceData *schema.ResourceData, meta interface{}) error {
+			orgProgUserWrapper := resourceObject.(*commons.OrgProgUserWrapper)
+			orgProgUser := orgProgUserWrapper.GetOrgProgUser()
+			var value []*administration.ProgPolicy = nil
+			if v, ok := resourceData.GetOk(string(Policies)); ok {
+				if policies, err := expandPolicies(v); err != nil {
+					return err
+				} else {
+					value = policies
+				}
+			}
+			orgProgUser.SetPolicies(value)
+			return nil
+		},
+		nil,
+	)
+
+	fieldsMap[Accounts] = commons.NewGenericField(
+		commons.AdministrationOrgProgrammaticUser,
+		Accounts,
+		&schema.Schema{
+			Type:     schema.TypeSet,
+			Optional: true,
+			ForceNew: true,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					string(AccountId): {
+						Type:     schema.TypeString,
+						Required: true,
+					},
+					string(AccountRole): {
+						Type:     schema.TypeString,
+						Required: true,
```

Comment:
re-check if it's required param or optional

Reply:
Required tested thru postman